catroid/src/main/java/org/catrobat/catroid/content/Look.java

```diff
@@ -377,6 +368,25 @@ public float getDirectionInUserInterfaceDimensionUnit() {
 
 	public void setRotationMode(int mode) {
 		rotationMode = mode;
+		rotateLookDatasIfNeeded(mode);
+	}
+
+	private void rotateLookDatasIfNeeded(int mode) {
+		if (mode != Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && sprite.look.isFlipped()) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(false);
+		}
+		boolean orientedLeft = sprite.look.getDirectionInUserInterfaceDimensionUnit() < 0;
+		if (mode == Look.ROTATION_STYLE_LEFT_RIGHT_ONLY && orientedLeft) {
+			flipAllLookDatas();
+			sprite.look.setFlipped(true);
+		}
+	}
+
+	public void flipAllLookDatas() {
+		for (LookData lookData : sprite.getLookDataList()) {
```

Comment:
I'm currently in a struggle. On the one hand I like your idea of flipping all LookDatas. You don't have to care about the flip state when changing the LookData. Flipping is also a really cheap operation. On the other hand it will immediately load *all* images into memory. This may block the UI thread for a while. Also moving the `flipAllLookDatas` and `rotateLookDatasIfNeeded` to the Sprite will create another dependency between Look and Sprite (e.g. by setDirectionInUserInterfaceDimensionUnit). :crying_cat_face:

Maybe I'm just overrating the loading problem because I really want to remove the Sprite reference from the Look so bad. I haven't even evaluated nor tested the result. Maybe it isn't a good idea to remove the Sprite anyway.

However, I would suggest to not flip all Looks at once. Just the current one. You also have to update the flip state when the [image is changed](https://github.com/Catrobat/Catroid/blob/310dc99d43f993f1f58adc4d997fce554e0f0754/catroid/src/main/java/org/catrobat/catroid/content/Look.java#L243). Since `Look.isFlipped` won't tell you reliably if the image is flipped, you've to use `TextureRegion.isFlipX` which is the better solution anyway.
What do you think?